File: frontend/src/types/scan.ts
export interface Biomarker {
  marker_name: string;
  value: string;
  numeric_value: number | null;
  category: string;
  is_flagged: boolean;
  reference_range: string | null;
}

export interface Scan {
  id: string;
  baby_id: string;
  status: string;
  confidence: number;
  biomarkers: Biomarker[];
  created_at: string;
  notes: string | null;
}

export interface AnalysisResponse {
  scan_id: string;
  status: string;
  confidence: number;
  biomarkers: Biomarker[];
  created_at: string;
}

export interface Baby {
  id: string;
  name: string;
  date_of_birth: string | null;
  notes: string | null;
  created_at: string;
}

File: backend/app/schemas/scan.py
"""Scan and biomarker response schemas."""

from pydantic import BaseModel


class BiomarkerResponse(BaseModel):
    marker_name: str
    value: str
    numeric_value: float | None
    category: str
    is_flagged: bool
    reference_range: str | None

    model_config = {"from_attributes": True}


class ScanResponse(BaseModel):
    id: str
    baby_id: str
    status: str
    confidence: float
    biomarkers: list[BiomarkerResponse]
    created_at: str
    notes: str | None

    model_config = {"from_attributes": True}
